fastvlm_json.py

```
@@ -0,0 +1,310 @@
+#!/usr/bin/env python3
+"""
+FastVLM JSON Output Demo
+
+This script demonstrates FastVLM with JSON-formatted output for better
+structured results, including descriptions and tags. It implements robust
+JSON validation and retry logic to ensure valid structured output.
+"""
+
+import os
+import sys
+import time
+import json
+import argparse
+import logging
+from pathlib import Path
+
+# Define exception classes for JSON handling
+class JSONParsingError(Exception):
```

Implemented proper exception handling with a JSONParsingError class as suggested. The code now raises a specific exception with full context when JSON parsing fails rather than returning a non-standard structure. This makes the API more predictable and consistent. Added test_json_error_handling.py to verify this behavior.